DOC_CHANGES.md

```diff
@@ -5,3 +5,35 @@ Example Doc Change:
 ### Headline for the required change
 Description of the required change.
 -->
+
+### Resources must now use `provides` to declare recipe DSL
+
+Resources declared in `Chef::Resource` namespace will no longer get recipe DSL
+automatically.  Instead, explicit `provides` is required in order to have DSL:
+
+```ruby
+module MyModule
+  class MyResource < Chef::Resource
+    provides :my_resource
+  end
+end
+```
+
+Users are encouraged to declare resources in their own namespaces instead of putting them in the special `Chef::Resource` namespace.
+
+The `Chef::Resource` namespace will no longer get automatic DSL in a future Chef, and will emit a deprecation warning in Chef 12.
```

Comment:
This could probably be re-worded a little better:

Authors of HWRPs need to be aware that in the future all resources and providers will be required to include a provides line.   Starting with Chef 12.4.0 any HWRPs that do not have provides lines will trigger deprecation warning messages to add them.  The LWRPBase code does this automatically so LWRP authors and users who write classes that inherit from LWRPBase do not need to explicitly include provides lines.

Or something like that.  Spell out who needs to pay attention to what you're doing by what they're doing (writing LWRPs/HWRPs/LHWRPs) and not by what you're doing (removing Chef::Resource magic)...